devito/ir/support/basic.py

```diff
@@ -534,6 +534,67 @@ def distance_mapper(self):
                 retval[d] = j
         return retval
 
+    @cached_property
```

Comment:
all these properties here are simply lifted from what is now a subclass of Relation, namely Dependence , so this is not new code